pycds/weather_anomaly.py

```diff
@@ -0,0 +1,162 @@
+#       The SQL SELECT statements were already written, and the work required to translate them to SQLAlchemy seemed
+#       unnecessary. See https://docs.sqlalchemy.org/en/latest/core/tutorial.html#using-textual-sql
+#       This decision subject to revision.
+
+# TODO: Factor out common subquery for non-discarded obs_raw_id's (as a view)
+# TODO: Factor out common query structure into a defined function (parameterized by min/max function [can this be done?]
+# and by cell_method
+
+class DailyMaxTemperature(Base, MaterializedViewMixin):
+    __selectable__ = text('''
+        SELECT
+            hx.history_id AS history_id,
+            obs.vars_id AS vars_id,
+            effective_day_for_Tmax(obs.obs_time, hx.freq) AS obs_day,
+            max(obs.datum) AS statistic,
+            sum(
+                CASE hx.freq
+                WHEN 'daily' THEN 1.0::float
+                WHEN '1-hourly' THEN (1/24.0)::float
+                WHEN '12-hourly' THEN 0.5::float
+                END
+            ) AS data_coverage
+        FROM
+            obs_raw AS obs
+            INNER JOIN meta_vars AS vars USING (vars_id)
+            INNER JOIN meta_history AS hx USING (history_id)
+        WHERE
+            obs.obs_raw_id IN (
+                -- Return id of each observation without any associated discard flags;
+                -- in other words, exclude observations marked discard, and don't be fooled by
+                -- additional flags that don't discard (hence the aggregate BOOL_OR's).
+                SELECT obs.obs_raw_id
+                FROM
+                    obs_raw AS obs
+                    LEFT JOIN obs_raw_native_flags  AS ornf ON (obs.obs_raw_id = ornf.obs_raw_id)
+                    LEFT JOIN meta_native_flag      AS mnf  ON (ornf.native_flag_id = mnf.native_flag_id)
+                    LEFT JOIN obs_raw_pcic_flags    AS orpf ON (obs.obs_raw_id = orpf.obs_raw_id)
+                    LEFT JOIN meta_pcic_flag        AS mpf  ON (orpf.pcic_flag_id = mpf.pcic_flag_id)
+                GROUP BY obs.obs_raw_id
+                HAVING
+                    BOOL_OR(COALESCE(mnf.discard, FALSE)) = FALSE
+                    AND BOOL_OR(COALESCE(mpf.discard, FALSE)) = FALSE
+            )
+            AND vars.standard_name = 'air_temperature'
+            AND vars.cell_method IN ('time: maximum', 'time: point', 'time: mean')
+            AND hx.freq IN ('1-hourly', '12-hourly', 'daily')
+        GROUP BY
+            hx.history_id, vars_id, obs_day
+    ''').columns(
+        column('history_id'),
+        column('vars_id'),
+        column('obs_day'),
+        column('statistic'),
+        column('data_coverage')
+    )
+    __primary_key__ = 'history_id vars_id obs_day'.split()
+
+class DailyMinTemperature(Base, MaterializedViewMixin):
+    __selectable__ = text('''
+        SELECT
+            hx.history_id AS history_id,
+            obs.vars_id AS vars_id,
+            date_trunc('day', obs.obs_time) AS obs_day,
```

Comment:
Also the `SELECT`s do not differ solely on the aggregate function; they also differ on the `WHERE` clause (specifically, `cell_method`). That said, they are very similar, I did ponder using a template (by which you mean a defined function, yes?). Perhaps that would be worthwhile. 

And perhaps it would be even better to formulate these as SQLAlchemy expressions; it might be easier to DRY these queries up that way. Thoughts?